### PR TITLE
[cleanup] added comments and cleaned around fixed point/lattice/reaching def

### PR DESCRIPTION
**Pull request by KevinHock**

### CODE REVIEW

File: pyt/fixed_point.py [hunk 1]
```diff
@@ -6,10 +6,10 @@ class FixedPointAnalysis():
     """Run the fix point analysis."""
 
     def __init__(self, cfg, analysis):
-        """Fixed point analysis
+        """Fixed point analysis.
 
-        analysis must be a dataflow analysis containing a 'fixpointmethod'
-        method that analyzes one CFG node"""
+        Analysis must be a dataflow analysis containing a 'fixpointmethod'
+        method that analyzes one CFG node."""
```
Comment (Thalmann) [inline pyt/fixed_point.py]: probably should fix "analyzes" here. It is with a z here and we have used british english in most of the project i am pretty sure :)

File: pyt/lattice.py [hunk 2]
```diff
@@ -6,35 +6,34 @@ def __init__(self, cfg_nodes, analysis_type):
         self.el2bv = dict()  # Element to bitvector dictionary
         self.bv2el = list()  # Bitvector to element list
         for i, e in enumerate(analysis_type.get_lattice_elements(cfg_nodes)):
+            # Give each element a unique shift of 1
             self.el2bv[e] = 0b1 << i
-            self.bv2el.append(e)
-        self.bv2el = list(reversed(self.bv2el))
+            self.bv2el.insert(0, e)
```
Comment (Thalmann) [inline pyt/lattice.py]: Make sure this is covered by the lattice unit tests :)

File: pyt/lattice.py [hunk 3]
```diff
@@ -6,35 +6,34 @@ def __init__(self, cfg_nodes, analysis_type):
...
 
     def get_elements(self, number):
-        r = list()
-
         if number == 0:
-            return r
+            return []
```
Comment (Thalmann) [inline pyt/lattice.py]: list()

File: pyt/lattice.py [hunk 4]
```diff
@@ -6,35 +6,34 @@ def __init__(self, cfg_nodes, analysis_type):
...
+        if constraint == 0b0:
+            return False
 
         try:
             value = self.el2bv[node1]
         except KeyError:
-            value = 0b0
-
-        if constraint == 0b0 or value == 0b0:
             return False
```
Comment (Thalmann) [inline pyt/lattice.py]: This should be fine, check if this method is unit tested.

**Comment by Thalmann** [comment]
Changing the analysis should not be in a cleanup commit. This is the vital part of PyT.
I will check out the analysis change more in depth at some later point.
Why did you change it?

File: pyt/fixed_point.py (hunk 1 above)
Comment (KevinHock) [inline pyt/fixed_point.py]: Good  catch 👍

**Comment by KevinHock** [comment]
I looked at this code pretty in depth when I did the trim command line flag. It's probably the coolest most important code in the codebase but it's harder to understand than the rest of it, because it's dataflow analysis. I think I kept everything semantically the same, right? I just made it a little easier to read IMO.

File: pyt/lattice.py (hunk 2 above)
Comment (KevinHock) [inline pyt/lattice.py]: I like that we have codeclimate now, it's so pretty https://codeclimate.com/github/python-security/pyt/coverage/59215a31fd706d000101e863

File: pyt/lattice.py (hunk 4 above)
Comment (KevinHock) [inline pyt/lattice.py]: 👍

File: pyt/lattice.py (hunk 3 above)
Comment (KevinHock) [inline pyt/lattice.py]: The other places in the codebase where we use `[]` are when we pass an empty list as an argument (in base_cfg.py and flask_adaptor.py) and one place where we do `while q != []:` in fixed_point.py. I kind of like `[]` when comparing, passing them as args and returning them as values. I completely agree that `list()` looks nicer during an assignment. Do you only disagree in the case of returning them as values? I don't have a strong opinion on this, but it's good to be on the same page as I wrote more code :)

Comment (Thalmann) [inline pyt/lattice.py]: Up to you :)

**Comment by KevinHock** [comment]
Nothing seems semantically different and it's been a month and a half so I'm just gonna merge, I'm sorry :/

**Pull request merged**